### PR TITLE
Ret 3396: Update Templates with Dynamic Links

### PR DESCRIPTION
**Pull request by fergusoshea**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RET-3396

### Change description ###
Update Email Templates to use a link that is defined in the backend rather than have multiple templates

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/et/syaapi/service/NotificationService.java [hunk 1]
```diff
@@ -213,6 +214,7 @@ public void sendAcknowledgementEmailToRespondents(CaseData caseData,
             emailToRespondentTemplate = notificationsProperties.getRespondentTseEmailTypeATemplateId();
         }
         respondentParameters.put("linkToDocument", Objects.requireNonNullElse(documentJson, ""));
+        respondentParameters.put("exuiCaseDetailsLink",notificationsProperties.getExuiCaseDetailsLink() + caseId);
```
Comment (JackReevies) [inline src/main/java/uk/gov/hmcts/reform/et/syaapi/service/NotificationService.java:217]: minor: comma

Comment (paul-peasegood) [inline src/main/java/uk/gov/hmcts/reform/et/syaapi/service/NotificationService.java:217]: Could we use String.format here? 

Other suggestion, could we write a method in notificationsPropties to generate a link, something like;
getExUiCaseLink(int caseId)

Comment (JackReevies) [reply src/main/java/uk/gov/hmcts/reform/et/syaapi/service/NotificationService.java:217]: If we've created `notificationsProperties.getExuiCaseDetailsLink() ` (ie, not lombok generated) then you could make that method take a parameter for the caseId

**Review by JackReevies — CHANGES_REQUESTED** [review]
Can we start naming these PRs in the format of "RET-XXXX: Short Description of Ticket or Ticket Title" as well please

File: src/main/java/uk/gov/hmcts/reform/et/syaapi/service/NotificationService.java (hunk 1 above)
Comment (fergusoshea) [reply src/main/java/uk/gov/hmcts/reform/et/syaapi/service/NotificationService.java]: It is a lombok class, so I went with the string.format option

File: src/main/java/uk/gov/hmcts/reform/et/syaapi/service/NotificationService.java [hunk 2]
```diff
@@ -326,6 +332,10 @@ public void sendResponseEmailToTribunal(
             HEARING_DATE,
             hearingDate
         );
+        tribunalParameters.put(
+            "exuiCaseDetailsLink",
+            notificationsProperties.getExuiCaseDetailsLink() + caseId
+        );
```
[the remark this answers is not in the document]
Comment (fergusoshea) [reply src/main/java/uk/gov/hmcts/reform/et/syaapi/service/NotificationService.java:324]: The tests haven't been written to check the parameter objects. That could be a good suggestion for another TD card.

File: src/main/java/uk/gov/hmcts/reform/et/syaapi/service/NotificationService.java [hunk 3]
```diff
@@ -148,6 +148,7 @@ public SendEmailResponse sendAcknowledgementEmailToClaimant(CaseData caseData,
         String emailToClaimantTemplate = TYPE_C.equals(claimantApplication.getContactApplicationType())
             ? notificationsProperties.getClaimantTseEmailTypeCTemplateId() :
             getAndSetRule92EmailTemplate(claimantApplication, hearingDate, claimantParameters);
+        claimantParameters.put("citizenPortalLink",notificationsProperties.getCitizenPortalLink() + caseId);
```
[the remark this answers is not in the document]
Comment (fergusoshea) [reply src/main/java/uk/gov/hmcts/reform/et/syaapi/service/NotificationService.java]: Done

File: src/main/java/uk/gov/hmcts/reform/et/syaapi/service/NotificationService.java (hunk 1 above)
Comment (fergusoshea) [reply src/main/java/uk/gov/hmcts/reform/et/syaapi/service/NotificationService.java]: Done

**Review by marcjudge — APPROVED** [review]

**Review by robin-ieong — APPROVED** [review]

**Review by JackReevies — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/et/syaapi/service/NotificationService.java [hunk 4]
```diff
@@ -284,12 +294,17 @@ public void sendAcknowledgementEmailToRespondents(CaseData caseData,
...
-        respondentParameters.put("shortText",
-                                 APP_TYPE_MAP.get(claimantApplication.getContactApplicationType()));
-        respondentParameters.put("datePlus7", LocalDate.now().plusDays(7).toString());
+        respondentParameters.put(
+            SEND_EMAIL_PARAMS_SHORTTEXT_KEY,
+            APP_TYPE_MAP.get(claimantApplication.getContactApplicationType())
+        );
+        respondentParameters.put(
+            SEND_EMAIL_PARAMS_DATEPLUS7_KEY,
+            LocalDate.now().plusDays(7).toString()
```
Comment (paul-peasegood) [inline src/main/java/uk/gov/hmcts/reform/et/syaapi/service/NotificationService.java:306]: LocalDate.now().plusDays(7).toString() is used twice can it be added to a varaible

**Pull request merged**